staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -637,6 +638,47 @@ func SearchMountPoints(hostSource, mountInfoPath string) ([]string, error) {
 	return refs, nil
 }
 
+// IsNotMountPoint determines if a directory is a mountpoint.
+// It should return ErrNotExist when the directory does not exist.
+// IsNotMountPoint is more expensive than IsLikelyNotMountPoint.
+// IsNotMountPoint detects bind mounts in linux.
+// IsNotMountPoint enumerates all the mountpoints using List() and
+// the list of mountpoints may be large, then it uses
+// isMountPointMatch to evaluate whether the directory is a mountpoint.
+func IsNotMountPoint(mounter Interface, file string) (bool, error) {
+	isMnt, sure, isMntErr := mountinfo.MountedFast(file)
+	notMnt := !isMnt
+	if sure == true && isMntErr == nil {
+		return !isMnt, nil
+	}
+	if isMntErr != nil && os.IsPermission(isMntErr) {
+		notMnt = true
+		isMntErr = nil
+	}
+	if isMntErr != nil {
+		return notMnt, isMntErr
+	}
+	// Resolve any symlinks in file, kernel would do the same and use the resolved path in /proc/mounts.
+	resolvedFile, err := filepath.EvalSymlinks(file)
```

Comment:
nit: what if a path is not absolute?

Reply:
It will resolve against from the current directory. Eg: https://go.dev/play/p/9CqUNqev4yX